docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-1339/README.md

```diff
@@ -0,0 +1,115 @@
+```
+
+## Compliant Solution (Integer)
+
+This compliant solution adheres more to standards by representing the account balance and item cost as `integers` in cents instead of using dollars. This approach eliminates the imprecision associated with floating-point numbers:
+
+*[compliant01.py](compliant01.py):*
+
+```py
+# SPDX-FileCopyrightText: OpenSSF project contributors
+# SPDX-License-Identifier: MIT
+""" Non-compliant Code Example """
+BALANCE = 300
+ITEM_COST = 33
+ITEM_COUNT = 5
+ 
+print(
+    f"{str(ITEM_COUNT)} items bought, ${ITEM_COST / 100} each. "
+    f"Current account balance: "
+    f"${str((BALANCE - ITEM_COUNT * ITEM_COST) / 100)}"
+)
+```
+
+This code produces the following output accurately:
+
+```text
+5 items bought, $0.33 each. Current account balance: $1.35
+```
+
+## Compliant Solution (Decimal)
+
+This compliant solution adheres to standards by utilizing the imported `Decimal` type, which allows for precise representation of decimal values. It's important to note that, on most platforms, calculations using `Decimal` tend to be less efficient compared to those using basic data types.
+
+*[compliant02.py](compliant02.py):*
+
+```py
+
+# SPDX-FileCopyrightText: OpenSSF project contributors
+# SPDX-License-Identifier: MIT
+""" Compliant Code Example """
+from decimal import Decimal
+BALANCE = Decimal("3.00")
+ITEM_COST = Decimal("0.33")
+ITEM_COUNT = 5
+ 
+print(
+    f"{str(ITEM_COUNT)} items bought, ${ITEM_COST} each. "
+    f"Current account balance: "
+    F"${str(BALANCE - ITEM_COUNT * ITEM_COST)}"
+)
+```
+
+This code outputs the following:
+
+```text
+5 items bought, $0.33 each. Current account balance: $1.35
+```
+
+## Automated Detection
+
+|Tool|Version|Checker|Description|
+|:----|:----|:----|:----|
+|[Pylint](https://pylint.pycqa.org/)|2023.10.1|Not Available|Not detected|
+
+## Related Guidelines
+
+|||
+|:---|:---|
+|[SEI CERT Oracle Coding Standard for Java](https://wiki.sei.cmu.edu/confluence/display/java/SEI+CERT+Oracle+Coding+Standard+for+Java?src=breadcrumbs)|[NUM04-J. Do not use floating-point numbers if precise computation is required](https://wiki.sei.cmu.edu/confluence/display/java/NUM04-J.+Do+not+use+floating-point+numbers+if+precise+computation+is+required)|
+|[The CERT C Secure Coding Standard](https://wiki.sei.cmu.edu/confluence/display/c/SEI+CERT+C+Coding+Standard)|[FLP02-C. Avoid using floating-point numbers when precise computation is needed](https://wiki.sei.cmu.edu/confluence/display/c/FLP02-C.+Avoid+using+floating-point+numbers+when+precise+computation+is+needed)
+|[The CERT C++ Secure Coding Standard](https://wiki.sei.cmu.edu/confluence/pages/viewpage.action?pageId=88046682) |[VOID FLP02-CPP. Avoid using floating point numbers when precise computation is needed](https://wiki.sei.cmu.edu/confluence/pages/viewpage.action?pageId=88046687)|
+|[ISO/IEC TR 24772:2010](http://www.aitcnet.org/isai/)| Floating-Point Arithmetic [PLF]|

```

Comment:
Last ISO/IEC link looks like spam. ISO is here:
https://www.iso.org/standard/61457.html 

but has issues, would go with the 2013 one as its to hard to figure out the correct super-seeded one.

Reply:
Fixed this link!